### ISSUE TITLE
Fan mode and oscillation mode

### ISSUE THREAD

**Issue by VespaChris** [igorxmath/homebridge-samsung-windfree-ac#9]
I really like your plug-in, it's working very well.

Do you plan to add fan mode and fan oscillation mode?

I am also happy to assist in adding the features.

Best regards,
Christian

**Comment by SpinninVinyl** [comment]
Agreed plugin works really well and super easy to setup. Tired four others prior to this one without success. 

Please Add the Fan/Dry Mode then its complete :-)

**Comment by micheledistefano79** [comment]
Yes, this plugin is amazing! One more vote to add the DRY mode. Thanks!

**Comment by igorxmath** [comment]
Both are implemented on `main` now, in 0145221 (#40), building on @RJLino's #31 and #32.

**Fan** — `OptionalFanControl` adds a `Fanv2` service with Active (on/off), CurrentFanState, TargetFanState (auto vs manual via `airConditionerFanMode`), and RotationSpeed mapping `low`/`medium`/`high`/`turbo` to 25/50/75/100%.

**Oscillation** — two ways, depending on what you want:
- `OptionalFanControl` includes a `SwingMode` on/off toggle on the fan service.
- `OptionalSwingDirectionSwitches` adds separate **Swing Vertical** and **Swing Horizontal** switches, for units that support directional oscillation.

Both are opt-in and gated on what the unit actually advertises (`airConditionerFanMode`, `fanOscillationMode`) — enabling one on a unit that doesn't support it logs a warning and leaves the control out, rather than adding a tile that shows "No Response".

On the follow-up comments asking for **Dry mode** (@SpinninVinyl, @micheledistefano79): that shipped separately as `OptionalDryModeSwitch` in 51acd74 (#38). HomeKit's `TargetHeatingCoolingState` has no "dry" value, so it's a switch — turning it off restores whatever mode the unit was in beforehand, tracked from the background poll so a mode set on the remote is restored too.

Not covered: the `wind` (fan-only) AC mode as its own switch, distinct from the fan speed controls above. If that's what you were after, worth a separate issue — happy to look at it.

Timing note: v1.2.0 was cut shortly before these merged, so they're on `main` but not yet on npm. Next release.

---
_Generated by [Claude Code](https://claude.ai/code)_
